partiql-cli/src/test/kotlin/org/partiql/cli/pico/InferCommandTest.kt

```diff
@@ -0,0 +1,362 @@
+import org.partiql.types.ListType
+import org.partiql.types.MissingType
+import org.partiql.types.NullType
+import org.partiql.types.SexpType
+import org.partiql.types.StaticType
+import org.partiql.types.StringType
+import org.partiql.types.StructType
+import org.partiql.types.SymbolType
+import org.partiql.types.TimeType
+import org.partiql.types.TimestampType
+import java.net.URL
+import java.time.Instant
+import java.util.stream.Stream
+import kotlin.test.Ignore
+
+class InferCommandTest {
+
+    private val userId = "TEST_USER"
+    private val command = InferCommand(listOf(LocalPlugin()))
+    private val catalogMap = listOf("aws", "b").associate { catalogName ->
+        val catalogUrl: URL = InferCommandTest::class.java.classLoader.getResource("catalogs/$catalogName") ?: error("Couldn't be found")
+        catalogName to ionStructOf(
+            field("connector_name", ionString("localdb")),
+            field("localdb_root", ionString(catalogUrl.path)),
+        )
+    }
+
+    companion object {
+        private const val DEFAULT_TABLE_NAME = "UNSPECIFIED"
+        const val CATALOG_AWS = "aws"
+        const val CATALOG_B = "b"
+        private val TYPE_AWS_DDB_PETS_ID = ValueDescriptor.TypeDescriptor(StaticType.INT)
+        private val TYPE_AWS_DDB_PETS_BREED = ValueDescriptor.TypeDescriptor(StaticType.STRING)
+        val TABLE_AWS_DDB_PETS = TableDescriptor(
+            name = DEFAULT_TABLE_NAME,
+            attributes = listOf(ColumnMetadata("id", TYPE_AWS_DDB_PETS_ID.type), ColumnMetadata("breed", TYPE_AWS_DDB_PETS_BREED.type))
+        )
+        val TABLE_AWS_DDB_B = TableDescriptor(
+            name = DEFAULT_TABLE_NAME,
+            attributes = listOf(ColumnMetadata("identifier", StaticType.STRING))
+        )
+        val TABLE_AWS_B_B = TableDescriptor(
+            name = DEFAULT_TABLE_NAME,
+            attributes = listOf(ColumnMetadata("identifier", StaticType.INT))
+        )
+        val TYPE_B_B_B_B_B = ValueDescriptor.TypeDescriptor(StaticType.INT)
+        private val TYPE_B_B_B_B = ValueDescriptor.TypeDescriptor(StructType(mapOf("b" to TYPE_B_B_B_B_B.type)))
+        val TYPE_B_B_B_C = ValueDescriptor.TypeDescriptor(StaticType.INT)
+        val TYPE_B_B_C = ValueDescriptor.TypeDescriptor(StaticType.INT)
+        val TYPE_B_B_B = ValueDescriptor.TypeDescriptor(
+            StructType(
+                fields = mapOf(
+                    "b" to TYPE_B_B_B_B.type,
+                    "c" to TYPE_B_B_B_C.type
+                )
+            )
+        )
+        // TODO: Need to throw errors on bad references
+        val TABLE_ERROR = TableDescriptor(DEFAULT_TABLE_NAME, attributes = listOf(ColumnMetadata("_1", StaticType.ANY)))
+    }
+
+    // TODO: Currently, SELECT a.a, a FROM <<{'a': 1}>> AS a isn't working, and it's causing issues. To remediate,
+    //  we need to add the input binding tuple to the PROJ operator's schema. This causes SELECT * FROM anything to fail.
+    @Ignore
```

Comment:
Those tests are ignored.

Reply:
ACK. Yes, this is because of the `*` issue. Once it's fixed, I'll remove the @Ignore annotation.